src/lib/temple/activity-new/fetch.ts

```diff
@@ -12,28 +13,54 @@ import type { DisplayableActivity, OperationsGroup } from './types';
 
 const LIQUIDITY_BAKING_DEX_ADDRESS = 'KT1TxqZ8QtKvLu3V3JH7Gx58n7Co8pgtpQU5';
 
+interface FetchActivitiesReturnValue {
+  activities: DisplayableActivity[];
+  reachedTheEnd: boolean;
+  oldestOperation?: TzktOperation;
+}
+
 export default async function fetchActivities(
   chainId: TzktApiChainId,
   account: TempleAccount,
   assetSlug: string | undefined,
   pseudoLimit: number,
   tezos: ReactiveTezosToolkit,
   knownBakersAndPayouts: TzktAlias[],
-  olderThan?: Activity
-) {
+  olderThan?: TzktOperation
+): Promise<FetchActivitiesReturnValue> {
   const operations = await fetchOperations(chainId, account, assetSlug, pseudoLimit, tezos, olderThan);
 
   const groups = await fetchOperGroupsForOperations(chainId, operations, olderThan);
 
   // TODO: replace filtering with transformation of liquidity baking activities into 'Unknown' activities
-  return groups
+  const activities = groups
     .map(({ operations }) =>
       parseTransactions(operations, account.publicKeyHash, knownBakersAndPayouts).filter(
         (activity): activity is DisplayableActivity =>
           activity.type !== ActivityType.LiquidityBakingBurn && activity.type !== ActivityType.LiquidityBakingMint
       )
     )
     .flat();
+  const reachedTheEnd = groups.length === 0;
+  const oldestOperation = groups.reduce<TzktOperation | undefined>(
+    (oldestOperationPrevGroups, group) =>
+      group.operations.reduce(
+        (oldestOperation, operation) =>
```

Comment:
reduce inside reduce doesn't look well readable.
And also naming is confusing for me (we use oldestOperation as variable name and as accumulator inside reduce